File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        // int count = 0;
        // int i = 0;
        // while(i+1<intervals.size()){
        //     if(intervals[i+1][0]<intervals[i][1]){
        //         //erase i+1;
        //         vector<vector<int>>::iterator it;
        //         it = intervals.begin();
        //         it+=i+1;
        //         intervals.erase(it);
        //         count++;
        //         continue;
        //     }
        //     i++;
        // }
        // return count;

        vector<vector<int>> ans;
        vector<int> temp = intervals[0];
        int i=1;
        while(i<intervals.size()){
            if(intervals[i][0]<temp[1]){
                if(intervals[i][1]<temp[1]){
                    temp = intervals[i];
                }
                i++;
            }
            else{
                ans.push_back(temp);
                temp = intervals[i];
                i++;
            }
        }
        return intervals.size() - ans.size() - 1;







    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a,b)->{
            return a[0]<b[0]?1:0;
        });
        int l = intervals[0][0];
        int r = intervals[0][1];

        int ans = 0;

        for(int i=1;i<intervals.length;i++){
            if(intervals[i][0]>=r){
                r = intervals[i][1];
                l = intervals[i][0];
            }
            else{
                if(intervals[i][1]<r){
                    r = intervals[i][1];
                    l = intervals[i][0];
                }
                ans++;
            }
        }
        return ans;
    }
}